sdks/java/io/redis/src/main/java/org/apache/beam/sdk/io/redis/RedisIO.java

```diff
@@ -0,0 +1,294 @@
+
+    /**
+     * The Redis commands related to read of key-value pairs.
+     */
+    public enum Command {
+      GET
+    }
+
+    @Nullable abstract RedisConnection connection();
+    @Nullable abstract Command command();
+    @Nullable abstract String keyPattern();
+    @Nullable abstract RedisService redisService();
+
+    abstract Builder builder();
+
+    @AutoValue.Builder
+    abstract static class Builder {
+      @Nullable abstract Builder setConnection(RedisConnection connection);
+      @Nullable abstract Builder setCommand(Command command);
+      @Nullable abstract Builder setKeyPattern(String keyPattern);
+      @Nullable abstract Builder setRedisService(RedisService redisService);
+      abstract Read build();
+    }
+
+    /**
+     * Define the connection to the Redis server.
+     *
+     * @param connection The {@link RedisConnection}.
+     * @return The corresponding {@link Read} {@link PTransform}.
+     */
+    public Read withConnection(RedisConnection connection) {
+      checkArgument(connection != null, "RedisIO.read().withConnection(connection) called with "
+          + "null connection");
+      return builder().setConnection(connection).build();
+    }
+
+    /**
+     * Define the Redis command to execute to retrieve the key/value pairs.
+     *
+     * @param command The Redis command.
+     * @return The corresponding {@link Read} {@link PTransform}.
+     */
+    public Read withCommand(Command command) {
+      checkArgument(command != null, "RedisIO.read().withCommand(command) called with null "
+          + "command");
+      return builder().setCommand(command).build();
+    }
+
+    /**
+     * Define the pattern to filter the Redis keys.
+     * @param keyPattern The filter key pattern.
+     * @return The corresponding {@link Read} {@link PTransform}.
+     */
+    public Read withKeyPattern(String keyPattern) {
+      checkArgument(keyPattern != null, "RedisIO.read().withKeyPattern(pattern) called with "
+          + "null pattern");
+      return builder().setKeyPattern(keyPattern).build();
+    }
+
+    /**
+     * Define the Redis service to use.
+     *
+     * @param redisService The {@link RedisService} to use.
+     * @return The corresponding {@link Read} {@link PTransform}.
+     */
+    public Read withRedisService(RedisService redisService) {
+      checkArgument(redisService != null, "RedisIO.read().withRedisService(service) called with"
+          + " null service");
+      return builder().setRedisService(redisService).build();
+    }
+
+    @Override
+    public void validate(PBegin input) {
+      checkState(connection() != null || redisService() != null, "RedisIO.read() requires a "
+          + "connection to be set withConnection(connection) or a service to be set "
+          + "withRedisService(service)");
+      checkState(command() != null,  "RedisIO.read() requires a command to be set via "
```

Comment:
It's weird to force users to do something about a parameter that has only one legal value. Just delete the enum and the getters/setters from the API.